velox/type/Type.h

```diff
@@ -888,6 +940,13 @@ struct TypeFactory<TypeKind::TIMESTAMP> {
   }
 };
 
+template <>
+struct TypeFactory<TypeKind::DATE> {
+  static std::shared_ptr<const DateType> create() {
+    return std::make_shared<DateType>();
```

Comment:
Let's use a single instance to avoid unnecessary making lots of them.

Reply:
This method is not needed after moving to ScalarType.